gateway/src/services/config-manager-v2.ts

```diff
@@ -98,7 +116,26 @@ export class ConfigurationNamespace {
       fs.readFileSync(this.#configurationPath, 'utf8')
     ) as Configuration;
     if (!this.#validator(configCandidate)) {
-      throw new Error(`Invalid configuration for namespace ${this.id}.`);
+      // merge with template file and try validating again
+      const configTemplateCandidate: Configuration = yaml.load(
+        fs.readFileSync(
+          path.join(ConfigTemplatesDir, path.basename(this.#configurationPath)),
+          'utf8'
+        )
+      ) as Configuration;
+      if (
+        !this.#validator(
+          Object.assign(configTemplateCandidate, configCandidate)
```

Comment:
As discussed in the 1:1 video conf, you need a deep copy here. A shallow copy like `Object.assign()` would overwrite the whole config tree as a single property - which will break new properties added in the template that are not defined at the top level.